docs/src/content/en/docs/agents/guardrails.mdx

```diff
@@ -0,0 +1,334 @@
+---
+title: "Guardrails | Agents | Mastra Docs"
+description: "Learn how to implement guardrails using input and output processors to secure and control AI interactions."
+---
+
+import { Tabs } from "nextra/components";
+
+# Guardrails
+
+Agents use processors to implement guardrails by intercepting, modifying, validating, or blocking messages at different stages of the AI interaction. Processors can transform, filter, or abort requests based on defined logic.
+
+Processors can be configured as:
+
+- **`inputProcessors`**: Applied before messages reach the language model.
+- **`outputProcessors`**: Applied to responses before they're returned to users.
+
+Some processors are *hybrid*, meaning they can be used with either `inputProcessors` or `outputProcessors`, depending on where the logic should be applied.
+
+## When to use processors
+
+Processors can be used for content moderation, prompt injection prevention, response sanitization, message transformation, and other security-related controls. Mastra provides several built-in input and output processors for common use cases.
+
+## Adding processors to an agent
+
+Import and instantiate the relevant processor class, and pass it to your agent’s configuration using either the `inputProcessors` or `outputProcessors` parameter:
+
+```typescript {3,9-17} filename="src/mastra/agents/moderated-agent.ts" showLineNumbers copy
+import { openai } from "@ai-sdk/openai";
+import { Agent } from "@mastra/core/agent";
+import { ModerationProcessor } from "@mastra/core/processors";
+
+export const moderatedAgent = new Agent({
+  name: "moderated-agent",
+  instructions: "You are a helpful assistant",
+  model: openai("gpt-4o-mini"),
+  inputProcessors: [
+    new ModerationProcessor({
+      model: openai("gpt-4.1-nano"),
+      categories: ["hate", "harassment", "violence"],
+      threshold: 0.7,
+      strategy: "block",
+      instructions: "Detect and flag inappropriate content in user messages",
+    })
+  ]
+});
+```
+
+<Tabs items={["Input", "Output","Hybrid",]}>
```

Comment:
I'm not sure `Tabs` is right here because the content is so long? It also seems to break the right-hand side nav 

The exception is if the `Tabs` are mutually exclusive (for example, a React guide and a Angular guide on the same page)